buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -462,14 +472,77 @@ private void logProcessInfo(String prefix, ProcessHandle.Info info) {
     }
 
     private void logFileContents(String description, Path from) {
-        logger.error("{} `{}`", description, this);
-        try(Stream<String> lines = Files.lines(from, StandardCharsets.UTF_8)) {
-            lines
-                .map(line -> "  " + line)
-                .forEach(logger::error);
+        final Map<String, Integer> errorsAndWarnings = new LinkedHashMap<>();
+        LinkedList<String> ring = new LinkedList<>();
+        try (LineNumberReader reader = new LineNumberReader(Files.newBufferedReader(from))) {
+            for (String line = reader.readLine(); line != null ; line = reader.readLine()) {
+                final String lineToAdd;
+                if (line.startsWith("[")) {
+                    // check to see if the previous message (possibly combined from multiple lines) was an error or
+                    // warning as we want to show all of them
+                    if (ring.isEmpty() == false) {
+                        String previousMessage = normalizeLogLine(ring.getLast());
+                        if (MESSAGES_WE_DONT_CARE_ABOUT.stream().anyMatch(match -> previousMessage.contains(match))) {
+                            ring.removeLast();
+                        } else {
+                            if (previousMessage.contains("ERROR") || previousMessage.contains("WARN")) {
+                                errorsAndWarnings.put(
+                                    previousMessage,
+                                    errorsAndWarnings.getOrDefault(previousMessage, 0) + 1
+                                );
+                            }
+                        }
+                    }
+                    lineToAdd = line;
+                } else {
+                    // We combine multi line log messages to make sure we never break exceptions apart
+                    if (ring.isEmpty()) {
+                        lineToAdd = line;
+                    } else {
+                        lineToAdd = ring.removeLast() + "\n" + line;
+                    }
+                }
+                if (MESSAGES_WE_DONT_CARE_ABOUT.stream().anyMatch(match -> lineToAdd.contains(match)) == false) {
```

Comment:
Couldn't we check the raw line for this first thing then we could shirt-circuit all the other stuff?

Reply:
I updated this logic to filter at the end. Raw line wouldn't work for stack-traces.